Compilation under 21.2.4 fails due to PhotonCalibrator

I'd like to use xAH with AnalysisBase 21.2.4, but when I try to compile I get the following errors. Apparently, EgammaInterfaces have migrated to a new location. 
Thanks!
Eric


```
In file included from /afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/xAODAnaHelpers/HelperFunctions.h:12:0,
                 from /afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:37:
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx: In member function 'virtual EL::StatusCode PhotonCalibrator::initialize()':
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:204:65: error: 'egammaPID' has not been declared
   ANA_CHECK( m_photonTightIsEMSelector->setProperty("isEMMask", egammaPID::PhotonTight));
                                                                 ^
/cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisBase/21.2.4/InstallArea/x86_64-slc6-gcc62-opt/src/Control/AthToolSupport/AsgTools/AsgTools/MessageCheck.h:353:23: note: in definition of macro 'ANA_CHECK'
   { const auto sc__ = EXP;      \
                       ^~~
/cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisBase/21.2.4/InstallArea/x86_64-slc6-gcc62-opt/src/Control/AthToolSupport/AsgTools/AsgTools/MessageCheck.h:354:47: error: template argument 1 is invalid
     typedef typename std::decay<decltype(sc__)>::type scType__;  \
                                               ^
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:204:3: note: in expansion of macro 'ANA_CHECK'
   ANA_CHECK( m_photonTightIsEMSelector->setProperty("isEMMask", egammaPID::PhotonTight));
   ^
/cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisBase/21.2.4/InstallArea/x86_64-slc6-gcc62-opt/src/Control/AthToolSupport/AsgTools/AsgTools/MessageCheck.h:355:10: error: 'isSuccess' is not a member of 'asg::CheckHelper<int>'
     if (!::asg::CheckHelper<scType__>::isSuccess (sc__)) {  \
          ^
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:204:3: note: in expansion of macro 'ANA_CHECK'
   ANA_CHECK( m_photonTightIsEMSelector->setProperty("isEMMask", egammaPID::PhotonTight));
   ^
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:205:66: error: 'egammaPID' has not been declared
   ANA_CHECK( m_photonMediumIsEMSelector->setProperty("isEMMask", egammaPID::PhotonMedium));
                                                                  ^
/cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisBase/21.2.4/InstallArea/x86_64-slc6-gcc62-opt/src/Control/AthToolSupport/AsgTools/AsgTools/MessageCheck.h:353:23: note: in definition of macro 'ANA_CHECK'
   { const auto sc__ = EXP;      \
                       ^~~
/cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisBase/21.2.4/InstallArea/x86_64-slc6-gcc62-opt/src/Control/AthToolSupport/AsgTools/AsgTools/MessageCheck.h:354:47: error: template argument 1 is invalid
     typedef typename std::decay<decltype(sc__)>::type scType__;  \
                                               ^
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:205:3: note: in expansion of macro 'ANA_CHECK'
   ANA_CHECK( m_photonMediumIsEMSelector->setProperty("isEMMask", egammaPID::PhotonMedium));
   ^
/cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisBase/21.2.4/InstallArea/x86_64-slc6-gcc62-opt/src/Control/AthToolSupport/AsgTools/AsgTools/MessageCheck.h:355:10: error: 'isSuccess' is not a member of 'asg::CheckHelper<int>'
     if (!::asg::CheckHelper<scType__>::isSuccess (sc__)) {  \
          ^
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:205:3: note: in expansion of macro 'ANA_CHECK'
   ANA_CHECK( m_photonMediumIsEMSelector->setProperty("isEMMask", egammaPID::PhotonMedium));
   ^
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:206:65: error: 'egammaPID' has not been declared
   ANA_CHECK( m_photonLooseIsEMSelector->setProperty("isEMMask", egammaPID::PhotonLoose));
                                                                 ^
/cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisBase/21.2.4/InstallArea/x86_64-slc6-gcc62-opt/src/Control/AthToolSupport/AsgTools/AsgTools/MessageCheck.h:353:23: note: in definition of macro 'ANA_CHECK'
   { const auto sc__ = EXP;      \
                       ^~~
/cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisBase/21.2.4/InstallArea/x86_64-slc6-gcc62-opt/src/Control/AthToolSupport/AsgTools/AsgTools/MessageCheck.h:354:47: error: template argument 1 is invalid
     typedef typename std::decay<decltype(sc__)>::type scType__;  \
                                               ^
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:206:3: note: in expansion of macro 'ANA_CHECK'
   ANA_CHECK( m_photonLooseIsEMSelector->setProperty("isEMMask", egammaPID::PhotonLoose));
   ^
/cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisBase/21.2.4/InstallArea/x86_64-slc6-gcc62-opt/src/Control/AthToolSupport/AsgTools/AsgTools/MessageCheck.h:355:10: error: 'isSuccess' is not a member of 'asg::CheckHelper<int>'
     if (!::asg::CheckHelper<scType__>::isSuccess (sc__)) {  \
          ^
/afs/cern.ch/user/e/ecorriga/private/DijetISR/ntuplemaking/DijetISRCode_Organizer/xAODAnaHelpers/Root/PhotonCalibrator.cxx:206:3: note: in expansion of macro 'ANA_CHECK'
   ANA_CHECK( m_photonLooseIsEMSelector->setProperty("isEMMask", egammaPID::PhotonLoose));
   ^
make[2]: *** [xAODAnaHelpers/CMakeFiles/xAODAnaHelpersLib.dir/Root/PhotonCalibrator.cxx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [xAODAnaHelpers/CMakeFiles/xAODAnaHelpersLib.dir/all] Error 2
make: *** [all] Error 2

```

--- a/.jenkins.yml
+++ b/.jenkins.yml
@@ -1,5 +1,5 @@
 env:
-  - ABV_RC=2.4.37 ABV_CM=21.2.3
+  - ABV_RC=2.4.37 ABV_CM=21.2.4
 
 before_script:
   - ls

--- a/Root/PhotonCalibrator.cxx
+++ b/Root/PhotonCalibrator.cxx
@@ -44,6 +44,9 @@
 
 #include <PATCore/PATCoreEnums.h>
 
+// apparently needed for egammaPIDs but was included in HelperClasses too?
+#include "ElectronPhotonSelectorTools/egammaPIDdefs.h"
+
 using HelperClasses::ToolName;
 
 // this is needed to distribute the algorithm to the workers

--- a/scripts/xAH_run.py
+++ b/scripts/xAH_run.py
@@ -257,7 +257,7 @@ if __name__ == "__main__":
 
     # at this point, we should import ROOT and do stuff
     import ROOT
-    if int(os.environ.get('ROOTCORE_RELEASE_SERIES', 0)) < 25:
+    if xAODAnaHelpers.utils.is_release20():
       xAH_logger.info("loading packages")
       ROOT.gROOT.Macro("$ROOTCOREDIR/scripts/load_packages.C")
     else:
@@ -284,7 +284,7 @@ if __name__ == "__main__":
     elif args.driver == 'condor':
       if getattr(ROOT.EL, 'CondorDriver') is None:
         raise KeyError('Cannot load the Condor driver from EventLoop. Did you not compile it?')
-      if not os.path.isfile(os.path.expandvars('$ROOTCOREBIN/../RootCore.par')):
+      if not os.path.isfile(os.path.expandvars('$ROOTCOREBIN/../RootCore.par')) and xAODAnaHelpers.utils.is_release20():
         raise IOError('I cannot find RootCore.par. Make sure you run `rc make_par` before running xAH_run.py.')
     elif args.driver == 'lsf':
       if getattr(ROOT.EL, 'LSFDriver') is None:

--- a/xAODAnaHelpers/Algorithm.h
+++ b/xAODAnaHelpers/Algorithm.h
@@ -283,7 +283,7 @@ namespace xAH {
 
             @endrst
          */
-        static std::map<std::string, int> m_instanceRegistry;
+        static std::map<std::string, int> m_instanceRegistry; //!
 
         /**
             @rst

--- a/xAODAnaHelpers/PhotonCalibrator.h
+++ b/xAODAnaHelpers/PhotonCalibrator.h
@@ -11,7 +11,11 @@
 
 // external tools include(s):
 #include "AsgTools/AnaToolHandle.h"
+#ifndef USE_CMAKE
 #include "PhotonEfficiencyCorrection/IAsgPhotonEfficiencyCorrectionTool.h"
+#else
+#include "EgammaAnalysisInterfaces/IAsgPhotonEfficiencyCorrectionTool.h"
+#endif
 #include "IsolationCorrections/IIsolationCorrectionTool.h"
 
 class AsgPhotonIsEMSelector;

